### PR TITLE
feat: use new community image

### PR DESCRIPTION
**Pull request by svor**
Replace dev image with new [Community Ansible Dev Spaces Image](https://github.com/ansible/ansible-workspace-env-reference-image/pkgs/container/ansible-workspace-env-reference)

screenshot-devspaces apps sandbox-m4 g2pi p1 openshiftapps com-2024 04 29-12_16_09

**Related issue:** https://issues.redhat.com/browse/CRW-6290

**To test current changes:**
1. Login into https://workspaces.openshift.com
2. Start a workspace by clicking the icon below
[Contribute](https://workspaces.openshift.com/f?url=https://github.com/devspaces-samples/ansible-devspaces-demo/tree/sv-new-community-image)
3. Check that all functionality works as expected

### CODE REVIEW

**Comment by svor** [comment]
cc @kyetter @dmytro-ndp @NilashishC

File: devfile.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ metadata:
 components:
   - name: tooling-container
     container:
-      image: quay.io/devspaces/ansible-creator-ee@sha256:a1d053d24f0b80438568bd4e06437900f0afe0ae29fcfa749bbace5cfff8c2ca
+      image: ghcr.io/ansible/community-ansible-devspaces:latest@sha256:42af3f305fd9b19d1493099723c95cba3127b67a0d703b75a7cae4e362ddfea6
```
Comment (ibuziuk) [inline devfile.yaml]: does ghcr.io has any pull limits ?

**Comment by bontreger** [comment]
I reject this change. The image size explodes from <900 MB to over 7.7 GB, causing significant issues with low-bandwidth connections. I understand the image is based off the universal developer image and that's an element we're wanting for supportability, but that container is only 5.7 GB, there's 2gb of unaccounted for "fluff" in the image that needs an explanation

**Comment by bontreger** [comment]
Can we rebase off the downstream UDI-rhel8? or explain the 2gb difference. We're dealing with some significant container-size bloat that causes performance and startup timeout issues on slower/throttled connections

image

**Review by ibuziuk — APPROVED** [review]

File: devfile.yaml (hunk 1 above)
Comment (NilashishC) [reply devfile.yaml]: Based on the comment [here](https://github.com/orgs/community/discussions/49671#discussioncomment-8795596) - it seems to not have one for public repos.

**Comment by svor** [comment]
> Can we rebase off the downstream UDI-rhel8? or explain the 2gb difference. We're dealing with some significant container-size bloat that causes performance and startup timeout issues on slower/throttled connections
> 
> image

@NilashishC could you please comment on this

**Comment by svor** [comment]
ghcr.io/ansible/community-ansible-devspaces:latest@sha256:42af3f305fd9b19d1493099723c95cba3127b67a0d703b75a7cae4e362ddfea6 image doesn't exist anymore, need to figure out

**Review by NilashishC — APPROVED** [review]

**Comment by NilashishC** [comment]
FYI: The latest SHA has changed. We did a new release 4 days back.


https://github.com/ansible/ansible-workspace-env-reference-image/pkgs/container/ansible-workspace-env-reference/211507328?tag=latest

**Comment by svor** [comment]
@NilashishC We can't use image with tag in Dev Spaces sample, it should be SHA that is not going to be change. Since https://github.com/ansible/ansible-workspace-env-reference-image/pkgs/container/ansible-workspace-env-reference/211507328?tag=latest provides only **latest** tag, we don't have stable SHA.

@dmytro-ndp As far as I remember we have some verification that all images contain SHA instead of tag in Dev Spaces, am I right?

**Comment by dmytro-ndp** [comment]
@svor 
> @dmytro-ndp As far as I remember we have some verification that all images contain SHA instead of tag in Dev Spaces, am I right?

you are right

To enable OpenShift image mirroring in an airgapped environment, it's necessary to use the sha256 hash in the image address.

**Comment by svor** [comment]
I'd like to get a review from @dmytro-ndp

**Comment by dmytro-ndp** [comment]
@svor : I have tested recommended extensions installation, devfile command run and ansible playbook code auto-completion in workspace created from https://github.com/devspaces-samples/ansible-devspaces-demo/tree/sv-new-community-image repo on workspaces.openshift.com, and in Dev Spaces 3.14.0.ER.

Everything worked as expected, except for one instance of workspace startup failure that I couldn't reproduce a second time: `Error creating DevWorkspace deployment: Container tooling-container has starte ImagePullBackOff`

Screenshot from 2024-05-09 15-14-15

**The note**: what do you think about enabling Ansible Lightspeed in Ansible extension by default, in order to avoid `Enable lightspeed services from settings to use the feature.` error when try to connect to Ansible Lightspeed:

screenshot-devspaces apps sandbox-m4 g2pi p1 openshiftapps com-2024 05 09-14_36_37

**Comment by svor** [comment]
@dmytro-ndp thank you for the review
> Error creating DevWorkspace deployment: Container tooling-container has starte ImagePullBackOff

I can't reproduce this problem on my side

>The note: what do you think about enabling Ansible Lightspeed in Ansible extension by default, in order to avoid Enable lightspeed services from settings to use the feature. error when try to connect to Ansible Lightspeed:

It's a great idea, issue for this improvement: https://issues.redhat.com/browse/CRW-6336

**Review by dmytro-ndp — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@svor : thanks for the heads up!

In fact, I couldn't reproduce start workspace error `Error creating DevWorkspace deployment: Container tooling-container has state ImagePullBackOff` on the same cluster or on another cluster.

**Pull request merged**